### PR TITLE
indexer: fix bug and optimize loading dynamic objects when calculating pool exchange rates

### PR DESCRIPTION
**Pull request by bmwill**
## Description 

indexer: fix bug and optimize loading dynamic objects when calculating pool exchange rates

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Review by stefan-mysten — APPROVED** [review]
Thanks for this fix @bmwill . I don't know enough yet about dynamic fields to understand some parts, so perhaps some other folks can have a look @wlmyng @oxade. Overall looks good to me. Once it's merged, I can try it out on the graphql side.

File: crates/sui-indexer/src/apis/governance_api_v2.rs
Comment (wlmyng) [inline crates/sui-indexer/src/apis/governance_api_v2.rs:239]: So if I'm understanding correctly, we were making an extraneous fetch? It seems like the data is already present and just needs to be deserialized from the df fetched from get_dynamic_fields_raw_in_blocking_task

File: crates/sui-indexer/src/apis/indexer_api_v2.rs
Comment (wlmyng) [inline crates/sui-indexer/src/apis/indexer_api_v2.rs:220]: i appreciate how explicit this is, thank you

File: crates/sui-indexer/src/indexer_reader.rs
Comment (wlmyng) [inline crates/sui-indexer/src/indexer_reader.rs:1178]: nice

File: crates/sui-indexer/src/models_v2/objects.rs
Comment (wlmyng) [inline crates/sui-indexer/src/models_v2/objects.rs:190]: neat, was worried we'd need to backfill the column to the correct data haha

**Review by wlmyng — APPROVED** [review]
awesome, tysm!!

File: crates/sui-indexer/src/apis/governance_api_v2.rs
Comment (bmwill) [reply crates/sui-indexer/src/apis/governance_api_v2.rs:239]: Yeah when we were doing the initial fetch we already had all the data

File: crates/sui-indexer/src/models_v2/objects.rs
Comment (bmwill) [reply crates/sui-indexer/src/models_v2/objects.rs:190]: yeah i was worried too, thankfully that's not the case!

File: crates/sui-indexer/src/models_v2/objects.rs
Comment (gegaowp) [inline crates/sui-indexer/src/models_v2/objects.rs:171]: wondering what's the trade-offs between storing the current `field_name` in DB, compared with the `bcs_name` this func returns below?

**Review by gegaowp — APPROVED** [review]
I do not have the most contexts but overall lgtm!

File: crates/sui-indexer/src/models_v2/objects.rs
Comment (bmwill) [reply crates/sui-indexer/src/models_v2/objects.rs:171]: That's a good question, and the answer is that its probably complicated. So today we store a DynamicFieldName which has both the type of the name and a json encode value of the key. This is the encoding that we end up sending on the wire back to clients, but we do also need the bcs form of it. Today to do that we do this complicated dance of looking at the type layout and then taking that and the json to convert back to bcs. I wonder if we could possibly side step that and go straight from json to bcs or figure out something else.

Comment (bmwill) [reply crates/sui-indexer/src/models_v2/objects.rs:171]: If you have any ideas or preferences please let me know!

**Pull request merged**
